### PR TITLE
[Terraform] new option "scopes" to define OAuth scopes for provider

### PR DESCRIPTION
**Pull request by soymo**
-----------------------------------------------------------------
# [all]
## [terraform]
New provider option `scopes` to define list of OAuth scopes for Google APIs
### [terraform-beta]
New provider option `scopes` to define list of OAuth scopes for Google APIs

### CODE REVIEW

File: third_party/terraform/utils/provider.go.erb [hunk 1]
```diff
@@ -61,6 +61,11 @@ func Provider() terraform.ResourceProvider {
 					"CLOUDSDK_COMPUTE_ZONE",
 				}, nil),
 			},
+            "scopes": &schema.Schema{
```
Comment (ndmckinley) [inline third_party/terraform/utils/provider.go.erb]: Although this will be fixed by codegen, can you indent this so it lines up with the rest of the code?

**Comment by soymo** [comment]
I like to use `access_token` in other provider (similar like in docs with kubernetes), but GKE requires access to user email. If there is no email in the scopes GKE returns an error and client_id in the place where client email should be, so my guess is GKE uses client_id instead of email. With this feature I can define list of scopes and when I add `https://www.googleapis.com/auth/userinfo.email` to current list it works fine.

**Comment by ndmckinley** [comment]
Ah, right - with the merge conflicts, I need you to rebase onto master and force-push up.  Once that's done, I'll merge.

**Comment by soymo** [comment]
@ndmckinley rebased and pushed.

**Pull request merged**